projects/packages/my-jetpack/src/class-initializer.php

```diff
@@ -318,11 +318,21 @@ public static function get_videopress_stats() {
 			);
 		}
 
-		$featured_stats = get_transient( self::VIDEOPRESS_STATS_KEY );
+		$featured_stats   = get_transient( self::VIDEOPRESS_STATS_KEY );
+		$videopress_stats = new VideoPress_Stats();
 
 		if ( ! $featured_stats ) {
-			$videopress_stats = new VideoPress_Stats();
-			$featured_stats   = $videopress_stats->get_featured_stats( 60 );
+			$featured_stats = $videopress_stats->get_featured_stats( 60, 'day' );
+		}
+
+		if ( is_wp_error( $featured_stats ) || ! $featured_stats ) {
+			return array(
+				'videoCount' => $video_count,
+			);
+		}
+
+		if ( $featured_stats['data']['views']['current'] < 500 || $featured_stats['data']['views']['previous'] < 500 ) {
+			$featured_stats = $videopress_stats->get_featured_stats( 2, 'year' );
```

Comment:
Hmmmm yes caching the period value for a week might be a good thing 🤔. If they've crossed the threshold into monthly or fallen back into yearly, I don't think they're going to care too much about not seeing that right away.

I'll see what I can do about structure, but I don't think it's possible to know if we need the yearly stats if we don't first check the monthly stats since we are basing that switch off the results of the monthly stats. Perhaps we can cache the stats a bit longer (6 hours? a day?) to avoid as many API calls

Reply:
I updated the structure and I think it works well. We can return early if the transient has data, and if the period transient exists, we can just retrieve that data and display it. If it doesn't, then we can check if the monthly data is substantial enough, and if it's not, get the yearly data. Thank you for the feedback!